querymonitor/querymonitor.pl

```diff
@@ -321,5 +332,108 @@ sub unixtime_to_printable{
 	}
 	die "ERROR: Unsupported format:'$format' ";
 }
+#------------------------------------------------------------------------------
+sub Process_the_CSV_through_Sqlite{
+	if ( -f $opt{ANALYZE} ){
+		# File exists - fall through and process it
+	}else{
+	    die "ERROR: 'ANALYZE' file does not exist: No file '$opt{ANALYZE}'.";	
+	}
+	print "Analyzing $opt{ANALYZE} ...\n";
+	
+	my ($sqlite_version) = do {my $vv=qx|$opt{SQLITE} --version|;chomp $vv;$vv=~/([\d\.]+)/};
+	$! and die "ERROR: Cannot run $opt{SQLITE} :$!";
+	$sqlite_version or die "ERROR: could not get SQLITE version";
+	# We do several extra steps to allow for OLD sqlite (< 3.8):
+	# (a) It does not support instr(), so we use xx like yy
+	# (b) the .import command does not allow --skip, so we pre-skip on input file
+	
+	# Use Forking and fifo to de-compress, remove first line, 
+	# then feed via the fifo into the ".import" command in sqlite
+	my $fifo = "/tmp/querymonitor_fifo_$$";
+	print "Creating temporary fifo $fifo ...\n";
+	qx|mkfifo $fifo|;
+	$! and die "ERROR: Fifo failed: $!";
+	
+	#Fork to provide CSV stream to fifo
+	my $extract_cmd = 'gunzip -c'; # expecting a gzipped file
+	$opt{ANALYZE} !~/\.gz$/i and $extract_cmd = 'cat'; # It is NOT gzipped 
+    my $pid = fork ();
+    if ($pid < 0) {
+      die "ERROR: fork failed: $!";
+    } elsif ($pid) {
+	  # This is the parent --fall through 
+    }else{
+		# This is the CHILD 
+	    #close std fds inherited from parent
+		close STDIN;
+		close STDOUT;
+		close STDERR;		
+		qx{$extract_cmd $opt{ANALYZE} | sed 1d > $fifo};
+		exit 0; # Exit child process and close FIFO
+	}
+	
+	if (! $opt{DB}){
+	   # DB name was not specified .. generate it from ANALYZE file name
+	   $opt{DB} = $opt{ANALYZE};
+	   $opt{DB} =~s/\.gz$//i;  # drop .gz
+	   $opt{DB} =~s/\.csv$//i; # drop .csv
+	   $opt{DB} .= ".sqlite";  # append .sqlite 
+	}
+	print "Creating sqlite database $opt{DB} ...\n";
+	
+	open my $sqlfh ,  "|-" , $opt{SQLITE} , $opt{DB} or die "ERROR: Could not run SQLITE";
+	print $sqlfh <<"__SQL__";
+.version
+.header off
+CREATE TABLE q (ts integer,clientHost,clientPort,elapsedMillis integer,id,keyspace,nodeName,privateIp,query,type);
+CREATE VIEW summary as select datetime((ts/600)*600,'unixepoch','-4 Hours') as EDT,
+    sum(case when query LIKE '% system.%'  then 1 else 0 end) as systemq,
+        sum(case when query LIKE '% system.%' then 0 else 1 end) as cqlcount,
+        sum(case when query LIKE '% system.%' and elapsedmillis > 120 then 1 else 0 end) as sys_gt120,
+        sum(case when query NOT LIKE '% system.%' and elapsedmillis > 120 then 1 else 0 end) as cql_gt120,
+        sum(case when query LIKE '% system.%' and nodeName
+            in ('yb-prod-CAS-PROD-Primary-n2','yb-prod-CAS-PROD-Primary-n3','yb-prod-CAS-PROD-Primary-n1','yb-prod-CAS-PROD-Primary-n5',
```

Comment:
As a sanity check, is this intentional that we're hard-coding in node names from customer?

Reply:
Reduced dependency on hard coded names in the next checkin.

The plan is  to capture Live Universe config (node/region/zone) into the file to be analyzed such that it does not need to be hard-coded.

Reply:
When this gets merged, it should probably be squashed. We don't want customer info in the git history if we can help it.

See: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges